web/src/types/steps.d.ts

```diff
@@ -0,0 +1,18 @@
+type Step = {
+  id: string;
+  title: string;
+  description: string;
+  completed: boolean;
+  current: boolean;
+};
+
+type StepProgressProps = {
+  steps: Step[];
+  isComplete?: boolean;
+  completionTitle?: string;
+  completionSubtitle?: string;
+  completionButtonText?: string;
+  completionButtonHref?: string;
+  introText?: string;
+  ariaLabel?: string;
+};
```

Comment:
These will be reused by the study management tasks, so I extracted them out here.